src/errorpage.h

```diff
@@ -287,10 +362,79 @@ class TemplateFile
      */
     bool tryLoadTemplate(const char *lang);
 
+    SBuf textBuf; ///< A Buffer to store the template
     bool wasLoaded; ///< True if the template data read from disk without any problem
     String errLanguage; ///< The error language of the template.
     String templateName; ///< The name of the template
     err_type templateCode; ///< The internal code for this template.
+    SBuf lastTemplateFile; ///< The last used path
+};
+
+/// Checks error pages text for syntax errors
+class ErrTextValidator {
+public:
+    ErrTextValidator() {}
+    ErrTextValidator(const char *aName) : name_(aName) {}
+
+    /// Setup the current object to handle the checked text as a configuration
+    /// file error page formatted string like those can be found in deny_info
+    /// configuration parameter.
+    /// \par filename the configuration file path
+    /// \par lineNo the number of parsed line
+    /// \par line the configuration file line
+    /// The parameters used to describe the error to the caller
+    /// \retval this
+    ErrTextValidator &useCfgContext(const char *filename, int lineNo, const char *line);
+
+    /// Setup the current object to handle checked text as a template error
+    /// page.
+    /// \retval this
+    ErrTextValidator &useFileContext(const char *templateFilename);
+
+    /// The debug level to use for debug messages
+    /// \retval this
+    ErrTextValidator &warn(int level) { warn_ = level; return *this; }
+
+    /// Throw on parse errors
+    /// \retval this
+    ErrTextValidator &throws() { onError_ = doThrow; return *this; }
+
+    /// Just report parse errors
+    /// \retval this
+    ErrTextValidator &report() { onError_ = doReport; return *this; }
+
+    /// Validate the passed text
+    /// \retval true if text validated false otherwise
```

Comment:
Thanks for explaining the return value, but I have a little remark about the term "validation" in this context. The ErrTextValidator class purpose, as described, "checking error pages text for syntax errors". Consequently, one would think that the "validated" text is the text lacking syntax errors. However, that is not true: as you explained in one of your previous comments, "validated" means that the outside code may use this template, regardless of its syntax errors. If my understanding is correct, could you please reflect this in the method description?

Reply:
You can consider a syntax error as fatal and do not validate a template or non fatal and validate the template depending the case. 

My english are not very good, however I think that the "validated" means that the template is valid and can be used. The caller should not care If it has errors and can be used, or it has not errors at all. The caller requested this behaviour though the ErrTextValidator.report(true) method.

We can add a doc like the following if you like : "May throw, report, or do not to validate (return false) on parse errors"

Reply:
I could suggest, for example:
```
\\\ Checks whether the text may be used in the caller context.
\\\ The successfully validated text does not mean syntax errors
\\\ absence: this depends on the error action configuration.
```

Use it, if you find it meaningful.